en/developer/frameworks/articles/search/03-queries-and-filters/02-building-search-queries-and-filters.markdown

```diff
@@ -193,52 +231,108 @@ user input. User-provided `keywords` would further refine the list of results
 that match the queries written here:
 
 ```java
-MatchQuery titleQuery = queries.match("title_en_US", "legal");
+package com.liferay.docs.search;
 
-TermsQuery rootFolderQuery = _queries.terms("folderId");
-rootFolderQuery.addValues("0");
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.search.document.Document;
+import com.liferay.portal.search.hits.SearchHit;
+import com.liferay.portal.search.hits.SearchHits;
+import com.liferay.portal.search.query.BooleanQuery;
+import com.liferay.portal.search.query.MatchQuery;
+import com.liferay.portal.search.query.Queries;
+import com.liferay.portal.search.query.TermsQuery;
+import com.liferay.portal.search.searcher.SearchRequest;
+import com.liferay.portal.search.searcher.SearchRequestBuilder;
+import com.liferay.portal.search.searcher.SearchRequestBuilderFactory;
+import com.liferay.portal.search.searcher.SearchResponse;
+import com.liferay.portal.search.searcher.Searcher;
 
-BooleanQuery booleanQuery = _queries.booleanQuery();
-booleanQuery.addMustQueryClauses(rootFolderQuery, titleQuery);
+import java.util.ArrayList;
+import java.util.List;
 
-SearchRequestBuilder searchRequestBuilder = 
-    _searchRequestBuilderFactory.getSearchRequestBuilder();
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
-searchRequestBuilder.emptySearchEnabled(true);
+@Component(
+	service = MySearchComponent.class
+)
+public class MySearchComponent {
 
-searchRequestBuilder.withSearchContext(
-            searchContext -> searchContext.setCompanyId(companyId);
-            searchContext -> searchContext.setKeywords(keywords);
-        );
+	public List<String> search(long companyId, String keywords)
+		throws PortalException {
 
-SearchRequest searchRequest = searchRequestBuilder.query(booleanQuery).build();
+		MatchQuery titleQuery = queries.match(
```

Comment:
Hey Tibor, Josh mentioned on the first PR (#67 iirc) that setting the same keywords into the search context and the match query didn't make sense.Since I see you doing that in this code sample,  I wanted to check your opinion on it in case it's not intentional here.

Reply:
:thinking: I will double check this tomorrow. I somehow lost/missed that feedback from Josh.